src/workflow/task.rs

```diff
@@ -1,34 +1,42 @@
-//! A [Closure] wrapped with [Config]uration, metadata, and so on.
+//! A [Closure] wrapped with Configuration, metadata, and optional secret.
+
 use crate::workflow::{closure::Closure, config::Resources};
-use libipld::Ipld;
+use libipld::{serde as ipld_serde, Ipld};
 use std::collections::BTreeMap;
 
+const RESOURCES_KEY: &str = "resources";
+const META_KEY: &str = "meta";
+const SECRETS_KEY: &str = "secret";
+
+const WITH_KEY: &str = "with";
+const DO_KEY: &str = "do";
+const INPUTS_KEY: &str = "inputs";
```

Comment:
👍 yeah good idea

Reply:
I feel like I'm converting between string types a lot 🤔 Is defaulting to `String` most places the wrong option?

Reply:
it's a Rust thing. Going w/ String, means no consts and literals anyway sadly.